sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -239,8 +266,8 @@ public URL toUrl() {
         if (this.snapshot != null) {
             url.setQueryParameter(Constants.UrlConstants.SNAPSHOT_QUERY_PARAMETER, this.snapshot);
         }
-        if (this.blobServiceSasQueryParameters != null) {
```

Comment:
Does anything use the `blobServiceSasQueryParameters` anymore? If someone upgraded to a new version of the Blobs SDK without updating to using `commonSasQueryParameters` would this continue to function?

Should this become an if/else block where `CommonSasQueryParameters` takes precedence over the Blob service variant? Such as:

```java
String encodedSas;
if (this.commonSasQueryParameters != null) {
    encodedSas = this.commonSasQueryParameters.encode();
} else if (this.blobServiceQueryParameters != null) {
    encodedSas = this.blobServiceQueryParameters.encode();
}

if (!CoreUtils.isNullOrEmpty(encodedSas)) {
    url.setQuery(encodedSas);
}
```

Reply:
With the other blob SAS PR the idea is that all the ServiceSasQueryParams classes are going away (once this PR merges in I can go in and deprecate the whole class essentially). Using BlobServiceSasSignatureValues will only work with Service SASs so it would technically only work in certain cases. 

I think the if/else block would just make it redundant since CommonSasQueryParameters is just a super set of BlobServiceSasQueryParameters

Reply:
Agreed with the change, my concern is the current production code could potentially begin failing upon upgrade without continuing to support `BlobServiceSasQueryParameters` to generate the query string. If I had the following code upon upgrade I could get a warning about using deprecated code but it would continue to compile, I would no longer have a URL which contains SAS query parameters given `CommonSasQueryParameters` is the only thing checked.

If we make sure to explicitly call out that they are still accepted in the API but won't trigger functionality I'm more okay with it.

```java
URL url = new BlobUrlParts()
  .setScheme("https://")
  .setHost("account.blob.core.windows.net")
  .setSasQueryParameters(blobServiceSasQueryParameters)
  .toUrl();
```